src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/SystemCommandLine/Utils.tt

```diff
@@ -522,6 +510,32 @@ public class OptionData
             }
         }
 
+        return options;
+    }
+
+    void GenerateCustomBinderClass(string className, string binderClassName, IReadOnlyCollection<OptionData> options)
+    {
+        if (className == null)
+        {
+            throw new ArgumentNullException(nameof(className));
+        }
+        if (binderClassName == null)
+        {
+            throw new ArgumentNullException(nameof(binderClassName));
+        }
+        if (options == null)
+        {
+            throw new ArgumentNullException(nameof(options));
+        }
+        if (className.Trim() == string.Empty)
+        {
+            throw new ArgumentException("Trimmed value cannot be empty", nameof(binderClassName));
```

Comment:
This error message requires the programmer to know that a trim was performed to result in an empty string. This could be more descriptive if phrased as "Value cannot be empty".

Reply:
Is it important to separate null from the whitespace check? Otherwise you can check for null or whitespace with String.IsNullOrWhiteSpace: https://docs.microsoft.com/en-us/dotnet/api/system.string.isnullorwhitespace?view=net-6.0

Reply:
Ok. Changed to string.IsNullOrWhiteSpace()